File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;

    int open =0, close = 0;
    for(int i=0; i< s.length(); i++)
    {
        if( s[i] == '(')
        {
            open++;
        }
        else
        {
            close++;
            if( close > open)
            {
                answer= false;
                break;
            }
        }
    }
    if( close != open)answer =false;
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;



class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        int cnt = 0;
        
        Stack<Integer> stack = new Stack<>();
        
        for( int i= 0; i< s.length(); i++){
            if(s.charAt(i) == ')' && cnt <= 0){
                answer = false;
                break;
            }
            else if( s.charAt(i) == '('){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        if( cnt != 0 ) answer= false;
                    
        return answer;
    }
}